backend/src/federation/index.ts

```diff
@@ -36,18 +52,47 @@ export const startDHT = async (
     logger.info(`keyPairDHT: publicKey=${keyPair.publicKey.toString('hex')}`)
     logger.debug(`keyPairDHT: secretKey=${keyPair.secretKey.toString('hex')}`)
 
+    const apiList = prepareCommunityApiList()
+    logger.debug(`ApiList: ${JSON.stringify(apiList)}`)
+
     const node = new DHT({ keyPair })
 
     const server = node.createServer()
 
     server.on('connection', function (socket: any) {
       // noiseSocket is E2E between you and the other peer
       // pipe it somewhere like any duplex stream
-      logger.info(`Remote public key: ${socket.remotePublicKey.toString('hex')}`)
+      logger.info(`server on... with Remote public key: ${socket.remotePublicKey.toString('hex')}`)
       // console.log("Local public key", noiseSocket.publicKey.toString("hex")); // same as keyPair.publicKey
 
-      socket.on('data', (data: Buffer) => logger.info(`data: ${data.toString('ascii')}`))
+      socket.on('data', async (data: Buffer) => {
+        logger.info(`data: ${data.toString('ascii')}`)
+        const json = JSON.parse(data.toString('ascii'))
+
+        if (json.apiVersions && json.apiVersions.length > 0) {
+          const communities = new Array<DbCommunity>()
+
+          for (let i = 0; i < json.apiVersions.length; i++) {
+            const apiVersion = json.apiVersions[i]
+            let community = await DbCommunity.findOne({
```

Comment:
can we do this without reading first? just writing into database and overriding if needed?
Maybe our ORM will be overwhelmed with this, but SQL wise:
`REPLACE INTO communities VALUES (a, b, c)`
Is this possible when there is an entry in the database with a fictional field d which we do not define but don't want to override when writing?

Reply:
I know about your requirement to use for the Insert/Update an optimized TypeORM solution.
i found these two interesting answers about this requirement:
- https://dev.to/rishit/optimizing-typeorm-tips-from-experience-part-1-dont-use-save-4ke9
- https://stackoverflow.com/questions/51583085/repository-save-uses-insert-instead-of-update

At the end the TypeORM needs a mechanism to check if the entity still exists to decide for insert or update.
This can on TypeORM level only be done by a leading read-access to have the correct entity-instances with their primary keys, After the attribut initializations calling save() the TypeORM can decide between insert and update base on the previous loaded entity isntances.
Please see the attached file for the SQL-Log for first INSERT and a follwong UPDATE per single save() invocation with a list of entities:
![grafik](https://user-images.githubusercontent.com/86960882/204659930-faeccee3-acad-475b-8a0a-176f221abc09.png)